### PR TITLE
Fix numerical stability of TorchLib erfcx

### LINKED ISSUE #1223: [torchlib] Fix special.erfcx
From https://github.com/microsoft/onnxscript/blob/1ff5388dc790e3e910a5a57fa5be70bcd3f650af/onnxscript/tests/function_libs/torch_lib/ops_test_data.py#L818

Tried to enable as
```python
TorchLibOpInfo("special.erfcx", special_ops.aten_special_erfcx),
```
but got redefine test error.

And based on https://pytorch.org/docs/master/special.html#torch.special.erfcx,
The implementation of special.erfcx should rather be:

```python
@torch_op("aten::special_erfcx")
def aten_special_erfcx(self: TFloatOrBFloat16) -> TFloatOrBFloat16:
    """special_erfcx(Tensor self) -> Tensor"""

    return op.Mul(op.Exp(op.Pow(self, 2)), op.Sub(1, op.Erf(self)))
```

This modification makes op test in converter side closer to result matching but still not enough for float32 precision. May need more investigations..

### PR DESCRIPTION
**Pull request by YusefSyed**
Fixes #1223.

The current `exp(x*x) * (1 - erf(x))` decomposition loses the small erf complement for positive inputs and can combine overflow with underflow. For example, the existing OpInfo inputs around 6.8 return zero instead of approximately 0.08. This change uses piecewise Cephes rational approximations, a reciprocal positive tail, and the negative-input reflection formula. It promotes half inputs for the calculation, preserves float64 constants, and masks inactive approximation inputs because ONNX `Where` evaluates both branches. Exported graphs need neither `Erf` nor `Loop`.

The existing erfcx xfail is removed without relaxing its tolerance. New tests cover interval boundaries, tails, special values, dtypes, shapes and empty inputs. The coefficient provenance and BSD notice are included in both wheel and source distributions.

Validation:

- Focused regression and existing OpInfo tests: 20 passed, 4 expected skips; rerun immediately before submission.
- 291,148 actual ONNX Runtime output comparisons, with graph optimizations disabled and enabled, including every float16 bit pattern and dense float32/float64 inputs.
- Independent SciPy checks and 180 high-precision mpmath reference values.
- Actual `torch.onnx.export` and ONNX Runtime execution passed for float32 and float64; lint, formatting, wheel and source-distribution checks passed.

```sh
.venv/bin/python -m pytest tests/function_libs/torch_lib/special_test.py tests/function_libs/torch_lib/ops_test.py -k erfcx -q
```

The runtime validation was on macOS arm64. BFloat16 dispatch is included but was not executed in these runtime checks; no GPU or throughput claim is made.

AI disclosure: OpenAI Codex assisted with the implementation, regression tests, validation tooling and this PR description. The numerical coefficients are adapted from the attributed SciPy/XSF source, not presented as newly derived approximations.

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/special.py [hunk 1]
```diff
@@ -15,13 +15,87 @@
...
+from onnxscript import ir
+from onnxscript.function_libs.torch_lib.ops import common as common_ops
 from onnxscript.function_libs.torch_lib.registration import torch_op
 from onnxscript.function_libs.torch_lib.tensor_typing import TFloat
 from onnxscript.onnx_opset import opset18 as op
 from onnxscript.onnx_types import TensorType
 
 _MATH_PI = math.pi
 
+# Coefficients adapted from SciPy XSF's Cephes ``ndtr.h`` (pinned source revision
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/special.py:27]: Thanks. Is there a version that comes from pytorch?

Comment (YusefSyed) [reply onnxscript/function_libs/torch_lib/ops/special.py:27]: Codex-assisted response: yes. PyTorch has Steven G. Johnson's [`erfcx_y100` / `calc_erfcx` implementation](https://github.com/pytorch/pytorch/blob/2e2b1371931ea2a080c423ee3bc15c4fc59787a6/aten/src/ATen/native/Math.h#L1707-L2187). It uses a 100-region polynomial table for the central positive range and a continued-fraction tail. The [CPU kernel calls it](https://github.com/pytorch/pytorch/blob/2e2b1371931ea2a080c423ee3bc15c4fc59787a6/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L619-L624); CUDA uses either [the corresponding Jiterator implementation or the same helper](https://github.com/pytorch/pytorch/blob/2e2b1371931ea2a080c423ee3bc15c4fc59787a6/aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu#L304-L317).

Those are different coefficients from the Cephes rational approximation used here. A PyTorch-sourced ONNX translation is feasible using coefficient lookup and Horner evaluation, but it would replace this approximation with the 700-coefficient table and its tail rules. I chose the current 42-coefficient rational form to avoid that lookup table in the export graph. Its SciPy provenance is explicit in the source/notice, and the focused erfcx tests still pass against PyTorch (20 passed, four expected skips). No performance comparison between the two ONNX representations has been established.

**Comment by YusefSyed** [comment]
Codex-assisted CI follow-up for the unchanged head `c679e0c`:

- The torch-nightly addbmm/baddbmm/logit/grouped-mm failures also occur in the corresponding [base CI run](https://github.com/microsoft/onnxscript/actions/runs/34295172174).
- Optional Lint fails during Docker `apt-get update` on expired Debian `bullseye-security` metadata, before running lint; the [base Optional Lint run](https://github.com/microsoft/onnxscript/actions/runs/34265697030) also fails.
- The Ubuntu py310 job passed its first pytest phase, then received a runner shutdown signal during the second phase. Its log does not show a test assertion causing that cancellation.
- The Windows onnx-ir-git conv2d failures did not reproduce in this [controlled Windows recheck](https://github.com/YusefSyed/onnxscript/actions/runs/34396242151): parent `d1c005d` and head `c679e0c` ran on the same machine, dependency environment and seed. Both passed the float16 conv2d checks. The head also passed its erfcx OpInfo checks and all 19 dedicated erfcx regressions.

The comparison artifacts include source/import identity, dependency freeze, JUnit and logs. It pins ONNX IR to `641d90447cac737fabd43d94ea8c1796bde3e3b4`; the original job reported version 1.1.0 without exposing its installed Git SHA, so this is not a claim to reproduce that entire environment exactly. No conv code or test tolerance was changed. The original Windows job still needs an upstream rerun to establish its current status; I do not have permission to rerun that upstream job.
